agent/engine/docker_task_engine_test.go

```diff
@@ -1720,3 +1720,170 @@ func TestHandleDockerHealthEvent(t *testing.T) {
 	})
 	assert.Equal(t, testContainer.Health.Status, api.ContainerHealthy)
 }
+
+func TestCreatedContainerMetadataUpdateOnRestart(t *testing.T) {
```

Comment:
I think you can combine all of these into a single test. Either a table based test or a task having all of these containers.